Add reply

Ryo-Fujimori opened the pull request:
返信機能までのPRです(解決できなかったエラーがあり、まだメール送信機能までは取り掛かっていません)

Ryo-Fujimori committed:
Update Post model
Add Parent ID

Ryo-Fujimori committed:
Update post model
Add parent connection

Ryo-Fujimori committed:
update post model

Ryo-Fujimori committed:
Update posts_controller
Add reply Action

Ryo-Fujimori committed:
Add replyform

Ryo-Fujimori commented:
原因が不明なエラー発生
場所は    = render 'posts/replyform', reply: @reply
の箇所
replyの初期化は行っているがエラーが解決されない

ActionView::Template::Error (no implicit conversion of nil into String)
Caused by: TypeError (no implicit conversion of nil into String)

Information for: ActionView::Template::Error (no implicit conversion of nil into String):
  
app/views/posts/show.html.haml:16

Information for cause: TypeError (no implicit conversion of nil into String):
![image](https://github.com/user-attachments/assets/47a2a1af-a526-4dce-a990-b846999d6825)

Ryo-Fujimori committed:
Update task

Ryo-Fujimori committed:
Add active_storage

bin/rails active_storage:install

Ryo-Fujimori committed:
create active_storage table

bin/rails db:migrate

Ryo-Fujimori committed:
Update post action

Ryo-Fujimori committed:
Update post form

Ryo-Fujimori committed:
Merge branch 'add-like' into Add-reply

Ryo-Fujimori committed:
Merge branch 'Add-reply' into update-post-add-image

uekkie commented on app/views/posts/_replyform.html.haml:
```diff
@@ -0,0 +1,5 @@
+  .form-inputs
+    = f.input :content, label: t('attributes.post.content') , class: 'form-control'
+  .form-actions
+    = f.submit t('save'), class: 'btn btn-primary',
```
https://github.com/Ryo-Fujimori/miniblog_4/pull/6#issuecomment-2869719450
このエラーの原因これっぽいです。
```suggestion
    = f.submit t('save'), class: 'btn btn-primary'
```

Ryo-Fujimori committed:
bug fix replyform

Ryo-Fujimori committed:
update posts spec

Ryo-Fujimori committed:
fix bug routes
ネストする位置が間違っていたので修正

Ryo-Fujimori committed:
fix replty action
@postが初期化されていなかったため修正

Ryo-Fujimori committed:
fix set_post action

ログイン中のユーザーからポストを取得していなかった、これではコントローラーを分ける意味がないので修正

Ryo-Fujimori committed:
update unfollow method

destroyアクションで使用するメソッドは例外を発生させたいので!を付ける

Ryo-Fujimori committed:
update before action

%i忘れ

Ryo-Fujimori committed:
使わないファイル

Ryo-Fujimori committed:
update user_params

require,permitは古いrailsの記法らしい、のでexpectを使用する

Ryo-Fujimori committed:
update message

英語メッセージだったので日本語に変更

Ryo-Fujimori committed:
冗長なコメントを削除

Ryo-Fujimori committed:
タイトルの日本語化

Ryo-Fujimori committed:
simpleformでユーザー編集画面を作成

Ryo-Fujimori committed:
rename followbutton

Ryo-Fujimori committed:
unlessを後ろに置いていた　→　みにくいので前置に変更

Ryo-Fujimori committed:
update relationship destroy action

destroyアクションでrelationのIDを渡さないのは設計的に不自然なのでunfolow!ではなくrelationshipのIDを渡すように変更

Ryo-Fujimori committed:
sgcopを追加

Ryo-Fujimori committed:
sgcopを実行

Ryo-Fujimori committed:
sgcopの結果を反映

bundle exec rubocop -A

Ryo-Fujimori committed:
メーラーの初期設定

Ryo-Fujimori committed:
日付の表示　strftime ⇒ l に変更

Ryo-Fujimori committed:
修正　new ⇒ build

Ryo-Fujimori committed:
ログアウト時の画面修正

Ryo-Fujimori committed:
テスト追加　user_path に遷移(未完成)

Ryo-Fujimori committed:
コンテンツモデルのルールに則って修正

https://www.perplexity.ai/search/htmlnokontentumoterunituitejie-ah0xUNAtQ622iMSSQqxF8Q#0

Ryo-Fujimori committed:
いいね、フォローボタンが表示されなくなっていたので修正

Ryo-Fujimori committed:
ja.ymlをモデルごとに分割するように変更

Ryo-Fujimori committed:
fix typo

Ryo-Fujimori committed:
名前変更　_replyform -> _reply_form

Ryo-Fujimori committed:
Merge pull request #4 from Ryo-Fujimori/add-User

Add user

Ryo-Fujimori committed:
コメント追加

Ryo-Fujimori committed:
Merge branch 'main' into Add-reply

Ryo-Fujimori committed:
返信機能修正

Ryo-Fujimori committed:
リプライ一覧追加
